packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -647,7 +647,10 @@ export function gid(id: string | undefined | null): string | null | undefined {
     if (isNull(id)) {
         return null;
     }
-    return `${id}-${vmBeingRendered!.idx}`;
+    if (isNull(vmBeingRendered) || hasShadow(vmBeingRendered)) {
```

Comment:
This null check is a bit odd; not sure if there's a more elegant way to do this in TypeScript. I'm pretty sure in this codepath the `vmBeingRendered` is never null.